balancer/weightedroundrobin/weightedroundrobin.go

```diff
@@ -19,11 +19,41 @@
 // Package weightedroundrobin defines a weighted roundrobin balancer.
 package weightedroundrobin
 
-// Name is the name of weighted_round_robin balancer.
-const Name = "weighted_round_robin"
+import "google.golang.org/grpc/attributes"
+
+const (
+	// Name is the name of weighted_round_robin balancer.
+	Name = "weighted_round_robin"
+
+	// Attribute key used to store AddrInfo in the Attributes field of
+	// resolver.Address.
+	attributeKey = "/balancer/weightedroundrobin/addrInfo"
+)
 
 // AddrInfo will be stored inside Address metadata in order to use weighted roundrobin
 // balancer.
 type AddrInfo struct {
 	Weight uint32
 }
+
+// AddAddrInfoToAttributes returns a new Attributes containing all key/value
+// pairs in a with ai being added to it.
+func AddAddrInfoToAttributes(ai *AddrInfo, a *attributes.Attributes) *attributes.Attributes {
```

Comment:
Should this take an address and return an address?

And this is not "add", this is "set", because it overrides the previous info, if there's one.

Reply:
Done.